src/test/java/net/openhft/chronicle/wire/issue/Issue739Test.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016-2022 chronicle.software
+ *
+ *       https://chronicle.software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.openhft.chronicle.wire.issue;
+
+import net.openhft.chronicle.bytes.Bytes;
+import net.openhft.chronicle.core.pool.ClassAliasPool;
+import net.openhft.chronicle.wire.*;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * see https://github.com/OpenHFT/Chronicle-Wire/issues/739
+ */
+public class Issue739Test extends WireTestCommon {
+
+    public static class One extends SelfDescribingMarshallable {
+        String text;
+
+        public One(String text) {
+            this.text = text;
+        }
+    }
+
+    public static class Two extends SelfDescribingMarshallable {
+        String text;
+
+        public Two(String text) {
+            this.text = text;
+        }
+    }
+
+    public static class Three extends SelfDescribingMarshallable {
+        private One one;
+        private Two two;
+        private One twoAndHalf;
+    }
+
+    public static class IThree extends SelfDescribingMarshallable {
+        private Marshallable one;
+        private Marshallable two;
+        private Marshallable twoAndHalf;
+
+    }
+
+    @Test
+    public void fieldAnchorAlias() {
+        Wire wire = new YamlWire(Bytes.wrapForRead(("three: !net.openhft.chronicle.wire.issue.Issue739Test$Three\n" +
+                "  one: &first\n" +
+                "    text: hello\n" +
+                "  two:\n" +
+                "    text: world\n" +
+                "  twoAndHalf: *first\n").getBytes()));
+        wire.getValueIn().typedMarshallable();
```

Comment:
This looks great!

Can we assert on the values of the demarshalled Three class?

Reply:
Done - please re-review.